### PR TITLE
Fix two problems with printing trees related to default arguments

### PR DESCRIPTION
**Pull request by mlangc**
Fixes [#1002564](https://www.assembla.com/spaces/scala-ide/tickets/1002564-rename-refactoring-is-broken-in-case-the-renamed-method-has-default-arguments).

### CODE REVIEW

File: src/main/scala/scala/tools/refactoring/sourcegen/ReusingPrinter.scala [hunk 1]
```diff
@@ -553,14 +606,30 @@ trait ReusingPrinter extends TreePrintingTraversals with AbstractPrinter {
...
+                rr
+              }
+            } else {
+              fun.pos match {
+                case pos: RangePosition =>
+                  // If fun is part of an expression involving default arguments, it might be
+                  // necessary to manually make sure that parenthesis and '.' is printed out.
+                  // See #1002564
+                  val parensAndSep = commentsAndSpaces ~ '(' ~ commentsAndSpaces ~ ')' ~ '.'.optional
+                  Layout(Movement.coveredStringStartingAtEndOf(pos, parensAndSep))
```
Comment (misto) [inline src/main/scala/scala/tools/refactoring/sourcegen/ReusingPrinter.scala]: This is very readable! Much better than any regex :+1:

**Comment by mlangc** [comment]
I've just committed some code that makes it possible to execute test cases in dedicated, unique packages without cluttering the test cases themselves... I think that this is better than adding unique packages by hand.

File: src/main/scala/scala/tools/refactoring/sourcegen/ReusingPrinter.scala (hunk 1 above)
Comment (mlangc) [inline src/main/scala/scala/tools/refactoring/sourcegen/ReusingPrinter.scala]: Thanks :-).. I think that there is an error though, as the optional `'.'` might be proceeded by `commentsAndSpaces`, as `foo /*strange*/  . /*???*/  bar()` is legal Scala. In fact, I'd like to go for something like `('(' ~ ')' ~ '.'.optional).skipping(commentsAndSpaces)`... it might make sense to do this in a separate PR though.

Comment (mlangc) [inline src/main/scala/scala/tools/refactoring/sourcegen/ReusingPrinter.scala]: I've just created a bug related to the limitation mentioned above: https://www.assembla.com/spaces/scala-ide/tickets/1002611
